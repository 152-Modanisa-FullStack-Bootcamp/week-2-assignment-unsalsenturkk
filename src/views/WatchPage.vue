<template>
  <div
      id="watchpage"
  >
    <h1>{{$route.query.id}}</h1>
  </div>
</template>

<script>
export default {
  name: "WatchPage",
}
</script>

<style scoped>

</style>