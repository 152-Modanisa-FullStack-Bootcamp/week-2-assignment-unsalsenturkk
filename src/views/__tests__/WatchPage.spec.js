import {createLocalVue, shallowMount} from "@vue/test-utils";
import WatchPage from "@/views/WatchPage";
import VueRouter from "vue-router";

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

const $route = {
    query: {id: 1}
}

describe("WatchPage.vue", () => {

    it('sanity test ', () => {
        return
    });

    describe("WatchPage unit test", () => {
        it('should component exists', () => {
            const wrapper = shallowMount(WatchPage, {
                localVue,
                router
            });

            expect(wrapper.exists()).toBeTruthy()
        });

        it('should component exists with id query', () => {
            const wrapper = shallowMount(WatchPage, {
                mocks: {
                    $route
                }
            });

            expect(wrapper.exists()).toBeTruthy()
        });
    })

    describe("WatchPage Snapshot", () => {
        it('renders WatchPage correctly ', () => {
            const wrapper = shallowMount(WatchPage, {
                mocks: {
                    $route
                }
            });
            expect(wrapper.element).toMatchSnapshot()
        });

    })
})