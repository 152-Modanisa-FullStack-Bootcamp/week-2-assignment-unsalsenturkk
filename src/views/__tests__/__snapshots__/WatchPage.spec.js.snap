// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchPage.vue WatchPage Snapshot renders WatchPage correctly  1`] = `
<div
  id="watchpage"
>
  <h1>
    1
  </h1>
</div>
`;
